homeassistant/helpers/trigger_template_entity.py

```diff
@@ -231,10 +232,26 @@ def _process_manual_data(self, value: Any | None = None) -> None:
         Ex: self._process_manual_data(payload)
         """
 
-        self.async_write_ha_state()
-        this = None
-        if state := self.hass.states.get(self.entity_id):
-            this = state.as_dict()
+        now = dt_util.utcnow()
+        attr = {}
+        if state_attributes := self.state_attributes:
+            attr.update(state_attributes)
+        if extra_state_attributes := self.extra_state_attributes:
+            attr.update(extra_state_attributes)
+
+        state = State(
```

Comment:
We need to construct out own state to provide an uptodate `this` variable in the template processing

Reply:
Oof. Can't we just not support `this` for trigger based template entities. It's pretty ugly to make a fake state.